src/NUnitConsole/nunit3-console.tests/ResultReporterTests.cs

```diff
@@ -46,12 +45,21 @@ public class ResultReporterTests
         [OneTimeSetUp]
         public void CreateResult()
         {
-            var mockAssembly = typeof (MockAssembly).Assembly;
-            var emptySettings = new Dictionary<string, object>();
+            var mockAssembly = typeof(MockAssembly).Assembly;
+            var emptySettings = new Dictionary<string, object>
+            {
+                { "TestParameters", "1=d;2=c" },
+                { "TestParametersDictionary", new Dictionary<string, string>
+                {
+                    { "1", "d" },
+                    { "2", "c" }
+                }}
+            };
 
-            var runner = new NUnitTestAssemblyRunner(new DefaultTestAssemblyBuilder());
-            runner.Load(mockAssembly, emptySettings);
-            var xmlText = runner.Run(TestListener.NULL, Framework.Internal.TestFilter.Empty).ToXml(true).OuterXml;
+            var controller = new FrameworkController(mockAssembly, "id", emptySettings);
+
+            controller.LoadTests();
+            var xmlText = controller.RunTests(null);
```

Comment:
Turns out the settings node is inserted at the FrameworkController level, and not the NUnitTestAssemblyRunner. To test this, I instead needed to move a level up in the framework API.

While doing that, I also removed the `Framework.Internal.TestFilter.Empty` reference - as we recently started supporting passing null or string.empty in it's place. Turns out that happened in a later version of the framework than the tests were currently using - hence the framework update.

Reply:
In fact, we should really stop using the framework API here. The console actually only deals with the engine but we ended up with code like this because I couldn't get a secondary engine to work alongside the "real" engine. That's due to the duplicate URI that arises when the secondary engine tries to initialize the `TestAgency` service.

In a different PR, I recently came up with a way to run a secondary `MasterTestRunner`, which is almost as good as a secondary engine. Alternatively, we could create code to configure a secondary engine that lacks the `TestAgency` and possibly some other services that are not needed.

I don't think that should be part of this PR, however, but something to look at in the future.

Reply:
That's a good point. I didn't think that far through.